packages/falso/src/lib/credit-card.ts

```diff
@@ -81,5 +86,9 @@ export function randCreditCard<Options extends CreditCardOptions = never>(
     };
   };
 
-  return fake(factory, options);
+  return fake(factory, options, { uniqueComparer: checkUnique });
+}
+
+function checkUnique(card: CreditCard, cards: CreditCard[]): boolean {
```

Comment:
This is repreated several times, maybe create a factory function that creates the chacker
```ts
function uniqueByKeys(keys){
  return (a, b) => objectIsUnique(keys)
}
```

Reply:
@shaharkazaz - Sorry I don't know how I missed this. Do you mean something like:

```ts
export function randCreditCard<Options extends CreditCardOptions = never>(
  options?: Options
) {
  const factory: () => CreditCard = () => {
    const fullName =
      options?.fullName ??
      `${randPersonTitle()} ${randFullName({ withAccents: false })}`;
    const brand = options?.brand ?? (randCreditCardBrand() as Brand);

    const dateOptions: Intl.DateTimeFormatOptions = {
      month: 'numeric',
      year: '2-digit',
    };
    const validFrom = randPastDate({ years: 1 }).toLocaleDateString(
      'en-GB',
      dateOptions
    );
    const untilEnd = randFutureDate({ years: 2 }).toLocaleDateString(
      'en-GB',
      dateOptions
    );

    return {
      fullName,
      brand,
      validFrom,
      untilEnd,
      ccv: randCreditCardCVV(),
      number: randCreditCardNumber({ brand }),
      account: randAccount(),
      type: rand(['Credit', 'Debit']),
    };
  };

  return fake(
    factory,
    options,
    {
      uniqueComparer: (card, cards) => objectIsUnique(card, cards, ['number'])
    }
  );
}
```